core/src/main/scala/akka/kafka/internal/SubSourceLogic.scala

```diff
@@ -59,7 +65,7 @@ private abstract class SubSourceLogic[K, V, Msg](
 
   /** We have created a source for these partitions, but it has not started up and is not in subSources yet. */
   var partitionsInStartup: immutable.Set[TopicPartition] = immutable.Set.empty
-  var subSources: Map[TopicPartition, Control] = immutable.Map.empty
+  var subSources: Map[TopicPartition, (Control, ActorRef)] = immutable.Map.empty
```

Comment:
It is not obvious which actor this is (source stage actor or consumer actor). I had to dig back and forth a while. Maybe a case class instead?

Reply:
It's the control of the `SubSourceStageLogic` and the stage actor for that stage.  I created a case class `SubSourceStageLogic.ControlAndStageActor`.